### DISCUSSION TITLE
[Support]: How do we securely make frigate available remotely?

### DISCUSSION THREAD

**Discussion by bobloadmire** [blakeblackshear/frigate#11885] [General Support]
### Describe the problem you are having

Frigate doesn't support any sort of authentication afaik. so if I forward it with something like cloudflare, its just open to the public? Whats the solution here?

### Version

latest

### Frigate config file

```yaml
n/a
```

### Relevant log output

```shell
n/a
```

### FFprobe output from your camera

```shell
n/a
```

### Operating system

Proxmox

### Install method

Docker Compose

### Coral version

Other

### Network connection

Wired

### Camera make and model

Tapo C120

**Comment by NickM-27** [comment]
frigate 0.14 does support authentication

**Reply by NickM-27** [reply]
however, it has always been an option to use something like cloudflare zero trust to require authentication (SSO based)

**Reply by banthungprong** [reply]
I still use these cloudflare tunnels for frigate 14/go2rtc and other services in my remote location. Only 1 account and password (Google or OTP) necessary. Much easier than handling different accounts/passwords.

**Reply by DrSpaldo** [reply]
Is there a way to disable anon access and limit user access with the new users?

**Comment by bakedhorse** [comment]
Pre-0.14, I always used Home Assistant to act as an authentication/interaction proxy for Frigate without directly exposing it to the internet, along with getting the benefits of the home automation.

However with the upcoming version, it should be much safer to expose frigate on the open internet with the built-in authentication login system. As long as you use port 8080 instead of port 5000.

**Reply by FaNt4zMa** [reply]
I'm sorry but I have my Home Assistant behind a reverse proxy but I'm unable to view my Frigate cameras from outside (LTE). Was this working for you? If so, how? Or do I need to expose Frigate too? Thanks

**Comment by DrSpaldo** [comment]
There are a few options in this thread:

https://www.reddit.com/r/frigate_nvr/comments/159cald/securing_your_frigate_installation_for_remote/

They are worth trying, I have done a few

**Comment by mimainka** [comment]
I'm currently using NGINX Proxy Manager and simply enable the Proxy Host when I need to access my Frigate from external where I don't have access to my VPN and deactivate it when I'm done.
Of course that's not authentication but rather limiting the time my Frigate is open to everyone knowing the URL to a minimum.

**Comment by jubjubrsx** [comment]
Not the best but works.. I offer nginx and set a password on the proxy.

**Comment by kingsmic62** [comment]
1. In Cloudflare create an application and point it at your Frigate server
2. Create a Cloudflare Tunnel to your application
3.. Create a policy for your application:
Name the policy > Allow > Select a Session duration
Add 2 rules:
   1. Include - Emails - your email
   2. Require - Login Methods - OTP

Save...

Next time you login to your Frigate instance it will be over ssl and you will be presented with the cloudflare login with otp - the pin will be sent to the email you specified in the policy above.

**Reply by hagak** [reply]
I thought cloudflare tunnels were not allowed with streaming content.  Wouldn't Frigate fall afoul to this?

**Reply by NickM-27** [reply]
they adjusted their ToS some time ago, regardless it likely depends on how much bandwidth, I have been using cloudflare for years without issues
